eZ/Publish/Core/REST/Server/Controller/User.php

```diff
@@ -179,16 +179,31 @@ public function loadUser($userId)
         $user = $this->userService->loadUser($userId);
 
         $userContentInfo = $user->getVersionInfo()->getContentInfo();
-        $userMainLocation = $this->locationService->loadLocation($userContentInfo->mainLocationId);
         $contentType = $this->contentTypeService->loadContentType($userContentInfo->contentTypeId);
 
+        $mainLocationId = $userContentInfo->mainLocationId;
+
+        // special case for loading current user, which does not have permissions to load its location.
+        if ($user == $this->repository->getCurrentUser()) {
```

Comment:
Ah, I didn't see the `==`, thought you where doing a `=`, please make the sentance more clear by doing:
`if ($user->id === $this->repository->getCurrentUser()->id) {`